src/main/java/io/cryostat/net/OpenShiftAuthManager.java

```diff
@@ -113,6 +133,29 @@ public AuthenticationScheme getScheme() {
         }
     }
 
+    @Override
+    public Optional<IntermediateResponse<UserInfo>> sendLoginRedirectIfRequired(
+            Supplier<String> headerProvider, Set<ResourceAction> resourceActions)
+            throws ExecutionException, InterruptedException {
+        Boolean hasValidHeader = false;
+        try {
+            hasValidHeader = this.validateHttpHeader(headerProvider, resourceActions).get();
+
+            if (hasValidHeader) {
```

Comment:
`hasValidHeader` is nullable since it is a `Boolean`, unfortunately, so this line could NPE if `validateHttpHeader` behaves badly. It would be safer to perform this check as `Boolean.FALSE.equals(hasValidHeader)`.